app/src/main/java/com/example/habitsmasher/ui/dashboard/RecyclerTouchListener.java

```diff
@@ -0,0 +1,754 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import android.animation.Animator;
+import android.animation.ObjectAnimator;
+import android.app.Activity;
+import android.graphics.Rect;
+import android.util.DisplayMetrics;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.VelocityTracker;
+import android.view.View;
+import android.view.ViewConfiguration;
+import android.view.animation.DecelerateInterpolator;
+import android.widget.ListView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Listener class that handles touch interactions with a RecyclerView
+ * Implementation of this class came from this source:
+ * Name: Velmurugan
+ * Date: March 4, 2021
+ * URL: https://howtodoandroid.com/android-recyclerview-swipe-menu
```

Comment:
Nice job finding this implemented online already. Let's try to go through this class one more time and try to change up the comments and explain each chunk of code better so it's not a direct copy/paste. Just have to add our own comments and maybe rearrange/rename some stuff.